File: src/main/java/ssg/search/collection/es/disp/ESBookItem.java
package ssg.search.collection.es.disp;

import com.google.common.collect.Lists;
import io.searchbox.core.SearchResult;
import ssg.search.parameter.Parameter;
import ssg.search.result.Item;
import ssg.search.result.Result;

import java.util.List;

public class ESBookItem extends ESItem {
	@Override
	public String getCollectionName(Parameter parameter) {
		return "book";
	}

	@Override
	public String getCollectionAliasName(Parameter parameter) {
		return "ESBookItem";
	}

	@Override
	public Result getResult(SearchResult searchResult, String name, Parameter parameter, Result result) {
		List<SearchResult.Hit<Item, Void>> list = searchResult.getHits(Item.class);
		List<Item> searchItemList = Lists.newArrayList();
		for (SearchResult.Hit h : list) {
			searchItemList.add((Item) h.source);
		}

		StringBuilder srchItemIds = new StringBuilder();
		super.processItemResult(parameter, searchItemList, srchItemIds);

		result.setBookItemList(searchItemList);
		result.setBookCount(searchResult.getTotal());
		result.setBookItemIds(srchItemIds.toString());
		return result;
	}
}

File: src/main/kotlin/ssg/front/search/api/collection/es/book/EsBookItem.kt
package ssg.front.search.api.collection.es.book

import org.elasticsearch.action.search.SearchResponse
import ssg.front.search.api.collection.es.common.EsItem
import ssg.front.search.api.dto.Parameter
import ssg.front.search.api.dto.result.Result

class EsBookItem : EsItem() {
    override var indexName = "book"

    override fun getResult(searchResponse: SearchResponse, name: String, parameter: Parameter, result: Result) {
        var (itemList, srchItemIds) = super.getItemResultPair(searchResponse, parameter)

        result.bookItemList = itemList
        result.bookCount = searchResponse.hits.totalHits.toInt()
        result.bookItemIds = srchItemIds
    }
}